Add cli refrences to the docs around annotations and labels

dfredell opened the pull request:
**What this PR does / why we need it**: Add links to the command line flags for enabling labels and annotations in the docs

As the docs sit today we had struggled with figuring out why the `kube_pod_labels` metric were not getting `label_POD_LABEL` to populate. When we read docs/pod-metrics.md, we got the impression that all LABELS would be scraped and should appear in the metrics. The kubernetes pod itself had the labels on it. 
We then stumbled upon  https://github.com/kubernetes/kube-state-metrics/issues/1489#issuecomment-851970288 and lead us to figure out that not all LABELS were extracted by default. Adding `--metric-labels-allowlist` solved our issue. So this MR is intended to make it more obvious when reading docs/pod-metrics.md that all labels are not scraped by default.

I only added the link to cli-arguments.md on docs that had the Description column. 

**How does this change affect the cardinality of KSM**:  does not change cardinality

**Which issue(s) this PR fixes** : fixes #1489

logicalhan commented:
/triage accepted
/assign @CatherineF-dev

fpetkovski reviewed (approved):
Thanks for the contribution 👍 

/lgtm

fpetkovski commented:
/hold for @CatherineF-dev to take a look

dfredell commented:
I presented/ reviewed this PR with my team and they caught some of my miss spellings, so I fixed, squashed, and pushed. 
`s/controled/controlled/ s/persistant/persistent/`

dgrisonnet commented:
Hi @dfredell, sorry for letting your PR go stale. The docs improvements that you are proposing look good to me, but it would be great if you could propagate them to all the docs referencing label/annotation metrics.

You can easily find those by running:
```
grep -rlE "_(labels|annotations)" docs
```

/remove-lifecycle stale
/unhold
/assign

dfredell commented:
@dgrisonnet  These docs tables were incredibly hard to read in a text editor without extra white space and formatting. So I through them through https://open-vsx.org/extension/yzhang/markdown-all-in-one 

I have added the Description column to those MDs that didn't have them as requested.

dgrisonnet commented:
@mrueg @rexagod any thoughts on this new format for the docs? I am personally fine with the change, but we will need to add a step to the lint to make sure it sticks.

mrueg commented:
We could probably run https://github.com/marketplace/actions/markdownlint-cli2-action if we wanted to to improve our markdown formatting.
